### ISSUE TITLE
[Feature Request] "Missed" from webserve/UI is different from missed in webapi/apibuilder

### ISSUE THREAD

**Issue by fernandog** [SiCKRAGETV/sickrage-issues#976]
- webapi /apibuild don't use the network setting local. Only network
- coming episodes javascript don't take into account the change local<>network and keep displaying that episode as missed

SR Version:	BRANCH: (develop) / COMMIT: (31217535710b3adeaab7bed65714553478ab18c5) 
SR User:	pi
SR Locale:	('pt_BR', 'UTF-8')
SR Config file:	/home/pi/SickRage/config.ini
SR Database file:	/home/pi/SickRage/sickbeard.db
SR Cache Dir:	/home/pi/SickRage/cache
SR Arguments:	['-q', '--daemon', '--nolaunch', '--pidfile=/var/run/sickrage/sickrage.pid', '--datadir=/home/pi/SickRage']
SR Web Root:	
Python Version:	2.7.3 (default, Mar 18 2014, 05:13:23) [GCC 4.6.3]

***Timezone setting: LOCAL***

**Only one episode in APIbuild**
```
/api/*****/?cmd=future&sort=date&type=missed
{
    "data": {
        "missed": [
            {
                "airdate": "2015-03-13", 
                "airs": "Friday 11:00 PM", 
                "ep_name": "Ectogenesis", 
            }
        ]
    }, 
    "message": "", 
    "result": "success"
```

**Two in coming episodes**
![1](https://cloud.githubusercontent.com/assets/2620870/6658705/c084babc-cb50-11e4-8a2f-a5f1812dee56.png)

***when changing timezone to NETWORK:***

coming episodes gets messy. Javascript problem.
apibulder return doesn't change.

![2](https://cloud.githubusercontent.com/assets/2620870/6658743/095d9104-cb52-11e4-8382-e5103c4927f8.png)

tried to clear browser CACHE and still don't work

@Hellowlol @Thraxis can you confirm? I know you uses a lot API.

**Comment by fernandog** [comment]
@TagForce can you confirm?

**Comment by abeloin** [comment]
For some reason the airdate is converted but not the airs. webapi.py#L764

**Comment by abeloin** [comment]
But this shouldn't change because of the network or local setting. The api should give the same time format for every query and maybe provide the timezone so the client can do the conversion.

**Comment by fernandog** [comment]
doing this will break compability with all third party apps.
Ir should convert the airdate/airs like the UI.
21:00 in London its not the same as 21:00 here. Thats why the local setting exists.

**Comment by abeloin** [comment]
Ok but then how do you know what time is shown via the api local or network?

**Comment by fernandog** [comment]
This same question is valid to coming episodes Depends on user preference: local or network. I dont need to know network airtime. If all my tv-shows are from my home country than its ok to know. but 100% are not.  dont you agree?

**Comment by abeloin** [comment]
I'm talking about the api, not the web interface. In the web interface, it's up to the user to choose it's preference. In the api, time for all tv shows should have a common ground, ie: zulu(GMT0) this way the software using the api can easily do the conversion to your local time(or not) without guessing if the user choose local or network

**Comment by Hellowlol** [comment]
I would like to choose the middle ground, add a new dict with airsUTC with full date, time and time zone to it can be parsed by third party apps and parameter for local time / network time with modifies the current time out put. Best for three worlds ;)

That's my 2 cents

**Comment by fernandog** [comment]
And if they dont want to handle it, just show whatever is in "airs"

Nzb360 dev wont handle this. They already told that.

**Comment by abeloin** [comment]
Sickbeard doesn't handle timezone in the api: https://github.com/midgetspy/Sick-Beard/blob/development/sickbeard/webapi.py#L789

**Comment by abeloin** [comment]
Ok according to this it is a bug: https://github.com/SiCKRAGETV/SickRage/commit/ae94ab6d80da23323324a9aaa0935c8bfe308eb5

All api should be using the time zone setting, so airs is incorrect

**Comment by abeloin** [comment]
Scratch everything I just said, it does work the shows I was testing where on tvrage with unrecognized tv station hence no timezone/no change and to other one is on YTV Japan but in tvdb it is ytv which default to YTV USA

**Comment by fernandog** [comment]
@midgetspy

is there a reason why UI missed concept is different com API missed ?

**UI:**
missed uses date and time

**API:**
uses only date

missed - show's date is older than today
http://sickbeard.com/api/#future

**Comment by fernandog** [comment]
@Hiltronix @MGaetan89  any of you want to try this? it's an old issue

**Comment by MGaetan89** [comment]
I can try. I will give you updates once I have some info.

**Comment by fernandog** [comment]
@MGaetan89 this will affects your app too =)

**Comment by MGaetan89** [comment]
It's ok, only the content changes, not the behavior.
If it helps make SickRage better, I'm fine with it :)

**Comment by fernandog** [comment]
I tried to convert it to consider time too but I couldn't

**Comment by MGaetan89** [comment]
I think this should be fixed as I work on #2567 
I will try to refactor this soon, right after I refactored the banner/fanart/poster part (as a practice, and because I work on that part already).

**Comment by OmgImAlexis** [comment]
We have switch to using feathub for feature requests. If this feature is still important to anyone, please
check that it has not been added and add it [HERE](http://feathub.com/SiCKRAGETV/SickRage)
Make sure to vote!
